### PR TITLE
feat: support lettuce cluster

### PR DESCRIPTION
**Pull request by DanLi39**

### CODE REVIEW

File: arex-instrumentation/redis/arex-redis-common/src/main/java/io/arex/inst/redis/common/lettuce/ReactiveStreamsConsumer.java [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+    }
+
+    /**
+     * 支持Mono类型的录制
+     * @param responseMono
+     * @return
+     */
+    public Mono<?> accept(Mono<?> responseMono) {
+        return responseMono
+            .doOnSuccess(o -> {
```
Comment (YongwuHe) [inline arex-instrumentation/redis/arex-redis-common/src/main/java/io/arex/inst/redis/common/lettuce/ReactiveStreamsConsumer.java:30]: result -> xxx

File: arex-instrumentation/redis/arex-redis-common/src/main/java/io/arex/inst/redis/common/lettuce/ReactiveStreamsConsumer.java [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+                traceTransmitter.transmit();
+                extractor.record(o);
+
+            })
+            .doOnError(o -> {
```
Comment (YongwuHe) [inline arex-instrumentation/redis/arex-redis-common/src/main/java/io/arex/inst/redis/common/lettuce/ReactiveStreamsConsumer.java:35]: throwable->xxx

File: arex-instrumentation/redis/arex-redis-common/src/main/java/io/arex/inst/redis/common/lettuce/RedisClusterReactiveResultUtil.java [hunk 3]
```diff
@@ -0,0 +1,43 @@
+package io.arex.inst.redis.common.lettuce;
+
+import io.arex.agent.bootstrap.model.MockResult;
+import io.arex.inst.redis.common.RedisExtractor;
+import reactor.core.publisher.Flux;
+import reactor.core.publisher.Mono;
+
+public  class RedisClusterReactiveResultUtil {
```
Comment (YongwuHe) [inline arex-instrumentation/redis/arex-redis-common/src/main/java/io/arex/inst/redis/common/lettuce/RedisClusterReactiveResultUtil.java:8]: why these methods no usage

File: arex-instrumentation/redis/arex-lettuce-v5/src/main/java/io/arex/inst/lettuce/v5/cluster/RedisClusterReactiveCommandsImplWrapper.java [hunk 4]
```diff
@@ -0,0 +1,499 @@
...
+    }
+
+    @Override
+    public Mono<Long> del(K... keys) {
+        return del(Arrays.asList(keys));
+    }
+
+    @Override
+    public Mono<Long> del(Iterable<K> keys) {
+        if (ContextManager.needReplay()) {
```
Comment (YongwuHe) [inline arex-instrumentation/redis/arex-lettuce-v5/src/main/java/io/arex/inst/lettuce/v5/cluster/RedisClusterReactiveCommandsImplWrapper.java:394]: not support ignoreMock

Comment (DanLi39) [reply arex-instrumentation/redis/arex-lettuce-v5/src/main/java/io/arex/inst/lettuce/v5/cluster/RedisClusterReactiveCommandsImplWrapper.java:394]: > not support ignoreMockI
IgnoreMock operation has been supported in io.arex.inst.redis.common.lettuce.RedisClusterReactiveResultUtil#monoReplay

File: arex-instrumentation/redis/arex-lettuce-v5/src/test/java/io/arex/inst/lettuce/v5/cluster/inst/RedisClusterClientInstrumentationTest.java [hunk 5]
```diff
@@ -0,0 +1,65 @@
+package io.arex.inst.lettuce.v5.cluster.inst;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+
+import io.arex.inst.redis.common.RedisConnectionManager;
+import io.lettuce.core.RedisURI;
+import io.lettuce.core.cluster.RedisClusterClient;
```
Comment (mr3) [inline arex-instrumentation/redis/arex-lettuce-v5/src/test/java/io/arex/inst/lettuce/v5/cluster/inst/RedisClusterClientInstrumentationTest.java:8]: Remove unused import statement

File: arex-instrumentation/redis/arex-lettuce-v5/src/main/java/io/arex/inst/lettuce/v5/cluster/RedisClusterClientExtractor.java [hunk 6]
```diff
@@ -0,0 +1,18 @@
...
+import java.util.concurrent.CompletableFuture;
+
+public class RedisClusterClientExtractor {
+
+    private RedisClusterClientExtractor() {}
+
+    public static <K, V> void addConnection(CompletableFuture<StatefulRedisClusterConnection<K, V>> connectionFuture, Iterable<RedisURI> redisURIs) {
+        connectionFuture.thenAccept(connection ->
+            // take first uri
+            RedisConnectionManager.add(connection.hashCode(), redisURIs.iterator().next().toString()));
```
Comment (mr3) [inline arex-instrumentation/redis/arex-lettuce-v5/src/main/java/io/arex/inst/lettuce/v5/cluster/RedisClusterClientExtractor.java:16]: Suggest use all redisURIs for computing in cluster mode

File: arex-instrumentation/redis/arex-lettuce-v6/src/main/java/io/arex/inst/lettuce/v6/cluster/RedisClusterClientExtractor.java [hunk 7]
```diff
@@ -0,0 +1,17 @@
...
+public class RedisClusterClientExtractor {
+
+    private RedisClusterClientExtractor() {}
+
+    public static <K, V> void addConnection(CompletableFuture<StatefulRedisClusterConnection<K, V>> connectionFuture, Iterable<RedisURI> redisURIs) {
+        connectionFuture.thenAccept(connection ->
+            // take first uri
+            RedisConnectionManager.add(connection.hashCode(), redisURIs.iterator().next().toString()));
+    }
+}
```
Comment (mr3) [inline arex-instrumentation/redis/arex-lettuce-v6/src/main/java/io/arex/inst/lettuce/v6/cluster/RedisClusterClientExtractor.java:17]: v5/v6.RedisClusterClientExtractor

Recommended to be merged into one class

**Review by mr3 — APPROVED** [review]

File: arex-instrumentation/redis/arex-redis-common/src/main/java/io/arex/inst/redis/common/lettuce/FluxConsumer.java [hunk 8]
```diff
@@ -0,0 +1,58 @@
...
+import io.arex.inst.runtime.util.TypeUtil;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.concurrent.atomic.AtomicInteger;
+import reactor.core.publisher.Flux;
+
+
+public class FluxConsumer {
+
+    private final RedisExtractor extractor;
```
Comment (lucas-myx) [inline arex-instrumentation/redis/arex-redis-common/src/main/java/io/arex/inst/redis/common/lettuce/FluxConsumer.java:18]: extraction universality:
```
// declare
private final BiConsumer extConsumer;
// call use
BiConsumer<RedisExtractor, FluxResult> extConsumer = (extractor, fluxResult) -> extractor.record(fluxResult);
```

**Review by lucas-myx — APPROVED** [review]

**Review by mr3 — DISMISSED** [review]

**Review by mr3 — APPROVED** [review]

**Review by lucas-myx — APPROVED** [review]

**Pull request merged**